src/navigator/navigation-bar/side-bar/side-bar-button/side-bar-button.tsx

```diff
@@ -52,7 +51,12 @@ export const SideBarButton: FC<Props> = ({
     if (disabled) {
       disabledOnPress();
     } else {
-      navigate(routeName);
+      if (onSwapButtonPress) {
+        onSwapButtonPress();
+      } else {
+        // @ts-ignore
+        navigate(routeName, params);
```

Comment:
Sorry, using `@ts-ignore` looks a bit shitty here. Please try finding a type or writing a new one like
```
interface NavigationArguments<T extends keyof NavigationParamList> {
  routeName: T;
  params: NavigationParamList[T];
}
```
then passing such value to this component and using it here. As an alternative, you may try using `NavigationArgsType`.